app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person in people:
+        if person["name"] not in Person.people:
+            person_init = Person(person["name"], person["age"])
+            Person.people[person["name"]] = person_init
+            person_list.append(person_init)
+    for person in people:
+        person_init = Person.people[person["name"]]
+        if "wife" in person.keys():
+            if person["wife"] is None:
+                person_init.wife = None
+            else:
+                person_init.wife = Person.people[person["wife"]]
+        if "husband" in person.keys():
```

Comment:
Instead of checking for the presence of a key using `in` and then accessing the value, it's better to use the `dict.get()` method. This approach simplifies the code and avoids potential `KeyError` exceptions. For example, you can use `person.get('wife')` instead of `if 'wife' in person.keys()`.